server/routes/replyRoutes.ts

```diff
@@ -33,18 +36,31 @@ router.get('/:id', async (req, res, next) => {
 
 })
 
-router.post('/', async (req, res, next) => {
+router.post('/', validateReply, async (req, res, next) => {
   try {
-    const reply: Reply = new Reply(req.body.user_id, req.body.post_id, req.body.message);
-    await getRepository(Reply).save(reply);
-    res.status(201).send(reply);
+    console.log(req.params)
+    const user = await getRepository(User).findOne(req.body.userId);
+    if(user) {
+      const post = await getRepository(Post).findOne(req.body.postId);
+      if(post) {
+        const reply: Reply = new Reply(user, post, req.body.message);
```

Comment:
No need for explicit typing

Reply:
In this case it seems to be necessary to check that the user nor the post are undefined and by checking the seperately it's easy to send out what's wrong! But I do agree that it makes the code look quite ugly, I'll come up with a more general solution!

Reply:
Lets look at this together in the session today 👍